pom.xml

```diff
@@ -185,20 +185,15 @@
     </dependency>
 
     <!-- other -->
-    <dependency>
-      <groupId>org.apache.maven.shared</groupId>
-      <artifactId>maven-dependency-tree</artifactId>
-      <version>3.2.1</version>
-    </dependency>
     <dependency>
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-utils</artifactId>
       <version>3.5.1</version>
     </dependency>
     <dependency>
-      <groupId>org.apache.maven.shared</groupId>
-      <artifactId>maven-artifact-transfer</artifactId>
-      <version>0.13.1</version>
+      <groupId>org.codehaus.plexus</groupId>
+      <artifactId>plexus-interpolation</artifactId>
+      <version>1.26</version>
     </dependency>
```

Comment:
@cstamas I use `maven-artifact-transfer` in one of the Maven plugins maintained by the Jenkins project and I was unaware that it was deprecated. If it is, what is its replacement and how should I migrate?

Reply:
@basil MAT was introduced to make transition from `org.sonatype.aether` Java package to `org.eclipse.aether` possible, and did it by use of Java reflection. This change happened when Aether (today Resolver) moved from Sonatype to Eclipse, between Maven 3.0 and Maven 3.1 (then it moved to ASF, but ASF got permission to use org.eclipse.aether package). The resolver packages will not be changed (for example to org.apache.maven.resolver) UNTIL it is also made "hidden" by Maven, and new Maven4 API will become the only way to interact with Maven.

This is to happen in far future, as similarly like Maven 3.x supported 2.x and 3.x plugins, Maven 4.x will support 4.x (the new API) and 3.x (current "API") plugins, hence the cut-oif is expected maybe in Maven 5, maybe Maven 4.5, still undecided (but also is safely far in future).

Hence, if you hopefully do NOT support Maven 3.0.x anymore (hopefully not, even ASF raised the bar to 3.2.5, that is still almost 10 years of history), all you have to do is to use Resolver classes "directly", just like this PR does.

Reply:
Can we release mat? I found it useful for code simplification

Reply:
OK, our minimum Maven version is 3.81 so I filed jenkinsci/maven-hpi-plugin#469. I appreciate your explanation. I have inherited several old Maven plugins and have been trying to modernize them, but it has been very challenging for me since I am an outsider to the Maven developer community and these transitions are not documented very well. That makes your explanation particularly helpful. Thanks!

Reply:
If you accept to not support Maven 3.0.x line (and anything older), it is totally safe to go for "direct use" of Resolver classes. Maven 4 introduces "new API" that will coexists with "old Maven 3.1+ API (am including resolver API here as well)" during Maven 4.x lifespan to offer transition period. It will be "cut off" (still undecided) probably in Maven 5, when the "new API" becomes the only way to interact with Maven, while all of it's internals and guts (unlike today, where you can access or inject just anything from Maven) becomes hidden from plugins.

Still, to be on safe side, please read this https://maven.apache.org/resolver/api-compatibility.html

Reply:
> Can we release mat? I found it useful for code simplification

Problem with MAT is that it uses Maven2 `ArtifactRepository` type for a local repository of "modern maven3 library" (org.sonatype or org.eclipse, does not matter). See issues:

https://issues.apache.org/jira/browse/MNG-7706
https://issues.apache.org/jira/browse/MNG-7663